### PR TITLE
[RFC] rimage: add functionality to resign fw image

### PR DESCRIPTION
**Pull request by juimonen**
Add function to resign already signed images. This can
be done with switch "q" as follows:

"rimage -q sof-tgl.ri -o sof-tgl-mod.ri -c tgl.toml -k key.pem"

Signed-off-by: Jaska Uimonen <jaska.uimonen@linux.intel.com>

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Will wait as discussed for the manifest check to be included in PR

**Review by lgirdwood — APPROVED** [review]

**Comment by marc-hb** [comment]
@juimonen  it looks like this PR added a lot of clang warnings (found by mere chance)

```
sof/rimage/src/manifest.c:1467:6: error: variable 'buffer' is used uninitialized whenever 'if' condition is true
      [-Werror,-Wsometimes-uninitialized]
        if (ret < 0) {
            ^~~~~~~
sof/rimage/src/manifest.c:1552:7: note: uninitialized use occurs here
        free(buffer);
             ^~~~~~
sof/rimage/src/manifest.c:1467:2: note: remove the 'if' if its condition is always false
        if (ret < 0) {
        ^~~~~~~~~~~~~~
sof/rimage/src/manifest.c:1459:6: error: variable 'buffer' is used uninitialized whenever 'if' condition is true
      [-Werror,-Wsometimes-uninitialized]
        if (size < 0) {
            ^~~~~~~~
sof/rimage/src/manifest.c:1552:7: note: uninitialized use occurs here
        free(buffer);
             ^~~~~~
sof/rimage/src/manifest.c:1459:2: note: remove the 'if' if its condition is always false
        if (size < 0) {
        ^~~~~~~~~~~~~~~
sof/rimage/src/manifest.c:1451:6: error: variable 'buffer' is used uninitialized whenever 'if' condition is true
      [-Werror,-Wsometimes-uninitialized]
        if (ret < 0) {
            ^~~~~~~
sof/rimage/src/manifest.c:1552:7: note: uninitialized use occurs here
        free(buffer);
             ^~~~~~
sof/rimage/src/manifest.c:1451:2: note: remove the 'if' if its condition is always false
        if (ret < 0) {
        ^~~~~~~~~~~~~~
sof/rimage/src/manifest.c:1438:14: note: initialize the variable 'buffer' to silence this warning
        void *buffer;
                    ^
                     = NULL
```

File: src/manifest.c [hunk 1]
```diff
@@ -1427,3 +1427,127 @@ int verify_image(struct image *image)
...
 }
+
+
+int resign_image(struct image *image)
+{
+	int key_size, key_file_size;
+	size_t size, read;
+	FILE *in_file;
+	void *buffer;
+	int ret, i;
```
Comment (marc-hb) [inline src/manifest.c:1438]: rimage does not live in the previous century like the kernel does, so you can move this at first point of use and combine declaration and initialization as allowed by C99

Comment (juimonen) [reply src/manifest.c:1438]: I guess all this goes away if just setting this to NULL at declaration as clang suggest... I mean it is a clear bug to free some bogus

Comment (juimonen) [reply src/manifest.c:1438]: @marc-hb made this: https://github.com/thesofproject/rimage/pull/78.

Comment (marc-hb) [reply src/manifest.c:1438]: Initializing all variables with some default value at the top makes sense... when and only when some default value actually makes sense. In other cases it merely hides compiler warnings finding bugs for C99 backward-compatibility (I didn't read that code in that PR).

**Pull request merged**